src/hazelcore/util/ListUtil.re

```diff
@@ -226,3 +226,16 @@ let take_while = (p: 'x => bool, xs: list('x)): list('x) =>
        [],
      )
   |> List.rev;
+
+let iterate = (f: 'x => ('y, 'x), times: int, start: 'x): (list('y), 'x) => {
```

Comment:
you should use an existing List helper function, e.g. fold_left, for this. for loops shouldn't be part of pure functional code.